rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableMetaCacheOracle.java

```diff
@@ -0,0 +1,293 @@
+/*
+ *  Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.alibaba.fescar.rm.datasource.sql.struct;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSetMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import com.alibaba.druid.pool.DruidDataSource;
+import com.alibaba.druid.util.StringUtils;
+import com.alibaba.fescar.common.exception.ShouldNeverHappenException;
+import com.alibaba.fescar.core.context.RootContext;
+import com.alibaba.fescar.rm.datasource.AbstractConnectionProxy;
+import com.alibaba.fescar.rm.datasource.DataSourceProxy;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+/**
+ * The type Table meta cache.
+ */
+public class TableMetaCacheOracle {
```

Comment:
This class is only used by the new undo manager, but the `BaseTransactionalExecutor` still uses the default `TableMetaCache`.
Additionally the `resultSetMetaToSchema` method already contains parts for "H2 drivers" and other stuff. So I would think of having additional Oracle stuff in the already existing `TableMetaCache` instead of inventing another class.

Reply:
我觉得还是在BaseTransactionalExecutor 这里加比较好。

Reply:
Ok, but why does this class then contain some "H2 driver" parts? I mean this class is for Oracle only, right?
(see line 281)